test/suites/optional/strings.py

```diff
@@ -19,6 +19,82 @@
                mountPath: /dev/myrenamedserial
 '''
 
+GDP_CONFIG_SERIAL_GLOB = '''
+genericDevicePlugin:
+  status: Enabled
+  devices:
+     - name: fakeserial
+       groups:
+         - paths:
+             - path: /dev/ttyUSB*
+'''
+
+POD_SERIAL_2_DEVICES = '''
+apiVersion: v1
+kind: Pod
+metadata:
+  name: serial-test-pod
+spec:
+  containers:
+  - name: serialdevice-app-container
+    image: registry.access.redhat.com/ubi9/ubi:9.6
+    command: ["sleep", "infinity"]
+    resources:
+      limits:
+        device.microshift.io/fakeserial: "2"
+    securityContext:
+      allowPrivilegeEscalation: false
+      capabilities:
+        drop: ["ALL"]
+      runAsNonRoot: true
+      seccompProfile:
+        type: "RuntimeDefault"
+'''
+
+POD_SERIAL_1_DEVICE = '''
+apiVersion: v1
+kind: Pod
+metadata:
+  name: serial-test-pod1
+spec:
+  containers:
+  - name: serialdevice-app-container
+    image: registry.access.redhat.com/ubi9/ubi:9.6
+    command: ["sleep", "infinity"]
+    resources:
+      limits:
+        device.microshift.io/fakeserial: "1"
+    securityContext:
+      allowPrivilegeEscalation: false
+      capabilities:
+        drop: ["ALL"]
+      runAsNonRoot: true
+      seccompProfile:
+        type: "RuntimeDefault"
+'''
+
+POD_SERIAL_2_DEVICES_SECOND = '''
+apiVersion: v1
+kind: Pod
+metadata:
+  name: serial-test-pod2
+spec:
+  containers:
+  - name: serialdevice-app-container
+    image: registry.access.redhat.com/ubi9/ubi:9.6
+    command: ["sleep", "infinity"]
+    resources:
+      limits:
+        device.microshift.io/fakeserial: "2"
+    securityContext:
+      allowPrivilegeEscalation: false
+      capabilities:
+        drop: ["ALL"]
+      runAsNonRoot: true
+      seccompProfile:
+        type: "RuntimeDefault"
+'''
```

Comment:
These three are basically identical. May I propose changing them to a function?

```suggestion
def get_ttyusb_pod_definition(name: str, num_devices: int):
    return f"""
apiVersion: v1
kind: Pod
metadata:
  name: {name}
spec:
  containers:
  - name: serialdevice-app-container
    image: registry.access.redhat.com/ubi9/ubi:9.6
    command: ["sleep", "infinity"]
    resources:
      limits:
        device.microshift.io/fakeserial: "{num_devices}"
    securityContext:
      allowPrivilegeEscalation: false
      capabilities:
        drop: ["ALL"]
      runAsNonRoot: true
      seccompProfile:
        type: "RuntimeDefault"
"""
```

Now, to obtain specific combination of name + amount of devices:

```
Get Ttyusb Pod Definition    serial-test-pod    2
Get Ttyusb Pod Definition    serial-test-pod1    1
Get Ttyusb Pod Definition    serial-test-pod2    2
```